src/coreclr/vm/i386/jitinterfacex86.cpp

```diff
@@ -961,32 +942,6 @@ void InitJITHelpers1()
     JIT_TrialAlloc::Flags flags = GCHeapUtilities::UseThreadAllocationContexts() ?
         JIT_TrialAlloc::MP_ALLOCATOR : JIT_TrialAlloc::NORMAL;
 
-    // Get CPU features and check for SSE2 support.
-    // This code should eventually probably be moved into codeman.cpp,
-    // where we set the cpu feature flags for the JIT based on CPU type and features.
-    int cpuFeatures[4];
-    __cpuid(cpuFeatures, 1);
-
-    DWORD dwCPUFeaturesECX = cpuFeatures[2];
-    DWORD dwCPUFeaturesEDX = cpuFeatures[3];
-
-    //  If bit 26 (SSE2) is set, then we can use the SSE2 flavors
-    //  and faster x87 implementation for the P4 of Dbl2Lng.
-    if (dwCPUFeaturesEDX & (1<<26))
-    {
-        SetJitHelperFunction(CORINFO_HELP_DBL2INT, JIT_Dbl2IntSSE2);
-        if (dwCPUFeaturesECX & 1)  // check SSE3
-        {
-            SetJitHelperFunction(CORINFO_HELP_DBL2UINT, JIT_Dbl2LngSSE3);
```

Comment:
Dbg2Long conversion have 1.6+x perf regression on my machine:
```
using System;
using System.Diagnostics;

for (;;)
{
    Stopwatch sw = new();
    sw.Start();
    Work(10_000_000_000);
    Console.WriteLine(sw.ElapsedMilliseconds.ToString());
}

static long Work(double iters)
{
    long sum = 0;
    for (double i = 0; i < iters; i += 42)
    {
        sum += (long)i;
    }
    return sum;
}
```

Reply:
I don't think this matters much since Intel people want to remove the asm implemementation anyway for the change to saturating, with MSVC they'll probably make the C++ helper call `_cvt_dtoll_sat` I guess. I agree with @tannergooding that any performance comparisons should be done when they're finished.

Reply:
Right, so it would be best to avoid changing and introducing regressions in the floating-point to integer conversions in this PR and leave all that for the Intel PR(s). Spreading changes like that into PRs with unrelated changes makes accounting more difficult.

Reply:
I've noticed something interesting, this seems to regress 25x on a CPU with AVX512, seems like there are some bad MSVC bugs here.

Reply:
I've restored the SSE3 FPU helper, SSE2 only CPUs should be so rare that it makes no sense to keep separate paths for them these days I'd say.